crates/services/p2p/src/peer_manager.rs

```diff
@@ -73,16 +74,23 @@ impl PeerManager {
         connection_state: Arc<RwLock<ConnectionState>>,
         max_non_reserved_peers: usize,
     ) -> Self {
+        let (reserved_peers_updates, _) = tokio::sync::broadcast::channel(1);
+
         Self {
             score_config: ScoreConfig::default(),
             non_reserved_connected_peers: HashMap::with_capacity(max_non_reserved_peers),
             reserved_connected_peers: HashMap::with_capacity(reserved_peers.len()),
             reserved_peers,
             connection_state,
             max_non_reserved_peers,
+            reserved_peers_updates,
         }
     }
 
+    pub fn reserved_peers_updates(&self) -> tokio::sync::broadcast::Sender<usize> {
```

Comment:
While that makes sense, I used `Sender` here since that was the (only?) way to exposes `Receiver`s from `SharedState` since `Receiver` is not cloneable.

Reply:
`Receiver` has `resubscribe` method=) 
<img width="704" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/636fd2a3-29b4-423f-b372-178073956e97">